lib/compress.js

```diff
@@ -87,6 +87,17 @@ function Compressor(options, false_by_default) {
         unused        : !false_by_default,
         warnings      : false,
     }, true);
+    var global_defs = this.options["global_defs"];
+    if (typeof global_defs == "object") for (var key in global_defs) {
+        if (/^@/.test(key)) {
```

Comment:
I don't bother with it normally as I don't expect people to pass in anything weird, but then I guess it's not a huge performance penalty so why not.